kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigMapWatcher.java

```diff
@@ -192,8 +199,18 @@ private void refreshEnvironment() {
         }
     }
 
+    /**
+     * Process {@link EventType#ERROR} events, restarting the watch when prior <em>resourceVersion</em> is gone.
+     *
+     * @see <a href="https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes">Efficient detection of changes</a>
+     */
     private void processConfigMapErrored(ConfigMapWatchEvent event) {
         LOG.error("Kubernetes API returned an error for a ConfigMap watch event: {}", event.toString());
+        if (event.getObject() instanceof Status && ((Status) event.getObject()).getCode() == HTTP_GONE) {
```

Comment:
How about extracting a variable here instead of calling `event.getObject()` twice?

Reply:
Done, thanks!